src/care_radiology/services/dicom_service.py

```
@@ -0,0 +1,219 @@
+
+class ServiceRequestLinkError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+        self.message = message
+
+
+def link_service_request_to_study(service_request_external_id, study_uid, raw_data=None):
+    try:
+        sr = ServiceRequest.objects.get(external_id=service_request_external_id)
+    except ServiceRequest.DoesNotExist:
+        raise ServiceRequestLinkError("No matching service request")
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)